CONTRIBUTING.md

```diff
@@ -0,0 +1,126 @@
+# Contributing
+
+## Design
+
+This project's development process is design-driven.
+Significant changes must be first discussed, and sometimes formally documented,
+before they can be implemented.
+
+This document describes the process for proposing, documenting, and implementing
+such changes. It's forked and modified from the Go language proposal process.
+
+### Goals
+
+- Make sure that proposals get a proper, fair, timely, recorded evaluation with
+  a clear answer.
+- Make past proposals easy to find, to avoid duplicated effort.
+- If a design doc is needed, make sure contributors know how to write a good one.
+
+### Definitions
+
+- A **proposal** is a suggestion filed as a GitHub issue, identified by having
+  the PROPOSAL title prefix.
+- A **design doc** is the expanded form of a proposal, written when the
+  proposal needs more careful explanation and consideration.
+
+### Scope
+
+The proposal process should be used for any notable change or addition.
+Since proposals begin (and will often end) with the filing of an issue, even
+small changes can go through the proposal process if appropriate.
+Deciding what is appropriate is matter of judgment we will refine through
+experience.
+If in doubt, file a proposal.
+
+### Process
+
+- Create an issue describing the proposal.
+
+- Like any GitHub issue, a Proposal issue is followed by an initial discussion
+  about the suggestion. For Proposal issues:
+  - The goal of the initial discussion is to reach agreement on the next step:
+    (1) accept, (2) decline, or (3) ask for a design doc.
+  - The discussion is expected to be resolved in a timely manner.
+  - If the author wants to write a design doc, then they can write one.
+  - A lack of agreement means the author should write a design doc.
+  - If there is disagreement about whether there is agreement,
+    the project's main commiter is the arbiter.
+
+- If a Proposal issue leads to a design doc:
+  - The design doc should be checked in to `design/NNNN-shortname.md`,
+    where `NNNN` is the GitHub issue number and `shortname` is a short name
```

Comment:
This feels like an obscene amount of process for contribution.  I would not contribute if I felt that it was going to involve so much bureaucracy.  If I ran into a production issue, I would fork, submit a PR with my change with a small explanation, and let the upstream handle it, but I would not spend more time than that on what should not be my problem any longer.